db/engine.go

```diff
@@ -83,8 +85,9 @@ func CheckEngine(dbPath string, createDatabaseIfNotExists bool, dbEngine Engine,
 	case EngineUnknown:
 		return EngineUnknown, ierrors.New("the database engine must not be EngineUnknown")
 
-	case EngineMapDB:
-		// no need to create or access a "database info file" in case of mapdb (in-memory)
+		// TODO: add an interface with a flag that indicates if the database needs the file system or not.
+	case EngineMapDB, EnginePostgreSQL:
+		// no need to create or access a "database info file" in case of mapdb (in-memory) or postgres (external database)
 		return EngineMapDB, nil
```

Comment:
this return is wrong if you pass EnginePostgreSQL

Reply:
damn :D